geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneServiceImpl.java

```diff
@@ -250,16 +250,21 @@ private void createIndexOnExistingRegion(PartitionedRegion region, String indexN
     LuceneIndexCreationProfile luceneIndexCreationProfile = new LuceneIndexCreationProfile(
         indexName, regionPath, fields, analyzer, fieldAnalyzers, serializer);
 
-    region.addCacheServiceProfile(luceneIndexCreationProfile);
-
+    region.acquireCacheServiceProfileUpdateLock();
     try {
-      validateLuceneIndexProfile(region);
-    } catch (IllegalStateException e) {
-      region.removeCacheServiceProfile(luceneIndexCreationProfile.getId());
-      throw new UnsupportedOperationException(
-          LocalizedStrings.LuceneIndexCreation_INDEX_CANNOT_BE_CREATED_DUE_TO_PROFILE_VIOLATION
-              .toString(indexName),
-          e);
+      region.addCacheServiceProfile(luceneIndexCreationProfile);
```

Comment:
Going with what Kirk mentioned, would it be better to have the addCacheServiceProfile method handle all of the locking?  You would possibly pass in some code/runnable to execute additional code before unlocking?